### PR TITLE
New App: Climate Clock

### PR DESCRIPTION
**Pull request by rgkimball**
This is a relatively simple, yet impactful app. We display two screens here relating to climate change:

- **The Deadline**: a countdown to the estimated point where a 1.5°C increase in global temperatures will be irreversible
- **The Lifeline**: the percent of the world's energy generated by renewable sources (wind, solar, hydro etc.)

NYC locals may recognize the first as the countdown that has displayed in Union Square above the Best Buy since 2020. Last year, similar clocks were also installed in Glasgow and Seoul. The origin story of this clock is related to Greta Thurnberg's UN speech and is intended to spread awareness about the urgency of climate change. Tidbyt owners who decide to display one or both of these clocks will have a chance to start a conversation with the people around them 

Much more information can be found about the clocks, underlying data and their proposals here: https://climateclock.world

## Screens

Although ClimateClock.world does offer a free API, the numbers can be easily calculated and their anchor points change infrequently. This app was therefore built with no external dependencies, and we can update the underlying data if our clocks become out of sync with the official ones.

The Lifeline

The Deadline

I don't consider myself to be much of an artist, so any enhancements to the graphics from the community are welcome!

There are also other numbers that could be implemented if people found them to be interesting and wanted to contribute graphics/PRs!

- AUM of the Green Climate Fund
- Land protected by indigenous people
- Climate change news feed (would require us to turn on the API, not terribly complicated)
- Other climate data not sourced from ClimateClock

### CODE REVIEW

**Comment by rgkimball** [comment]
I'm not sure the linter has it right here. If I make the suggested changes I'll be breaking the values:

```
                 "lang": "en",
-                "rate": 2.0428359571070087e-8,
-                "resolution": 1e-9,
+                "rate": 2.0428359571070087e - 8,
+                "resolution": 1e - 9,
                 "timestamp": "2020-01-01T00:00:00+00:00",
```

File: apps/climateclock/climate_clock.star [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+                "flavor": "lifeline",
+                "growth": "linear",
+                "initial": 11.4,
+                "labels": [
+                    "WORLD'S ENERGY FROM RENEWABLES",
+                    "GLOBAL RENEWABLE ENERGY",
+                    "RENEWABLES",
+                ],
+                "lang": "en",
+                "rate": 2.0428359571070087e-8,
```
Comment (betterengineering) [inline apps/climateclock/climate_clock.star:215]:
```suggestion
                "rate": "2.0428359571070087e-8",
```

File: apps/climateclock/climate_clock.star [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+
+def renewables(DATA):
+    fps = 20
+
+    data = DATA["data"]["modules"]["renewables_1"]
+    initial = data["initial"]  # 11.4
+    units = data["unit_labels"][0]
+    rate = data["rate"]
+    start = time.parse_time(data["timestamp"])
+    resolution = int(str(data["rate"])[-1]) + 1
```
Comment (betterengineering) [inline apps/climateclock/climate_clock.star:276]:
```suggestion
    resolution = int(data["rate"][-1]) + 1
```

File: apps/climateclock/climate_clock.star [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+                "flavor": "lifeline",
+                "growth": "linear",
+                "initial": 9.52,
+                "labels": [
+                    "GREEN CLIMATE FUND",
+                    "CLIMATE FUND",
+                    "GCF",
+                ],
+                "lang": "en",
+                "rate": 0,
```
Comment (betterengineering) [inline apps/climateclock/climate_clock.star:94]:
```suggestion
                "rate": "0",
```

File: apps/climateclock/climate_clock.star [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+                "flavor": "lifeline",
+                "growth": "linear",
+                "initial": 43.5,
+                "labels": [
+                    "LAND PROTECTED BY INDIGENOUS PEOPLE",
+                    "INDIGENOUS PROTECTED LAND",
+                    "INDIGENOUS PROTECTED",
+                ],
+                "lang": "en",
+                "rate": 0,
```
Comment (betterengineering) [inline apps/climateclock/climate_clock.star:114]:
```suggestion
                "rate": "0",
```

**Review by betterengineering — APPROVED** [review]
The linter is wrong, but we don't have a great way to tell it it's wrong 😅 . What about making it a string? It sucks since that means we'd have to make sure it's a string when updated. But the benefit is the linter won't accidentally change this line in the future.

**Comment by betterengineering** [comment]
I looked into simply ignoring these lines, hoping we could just add a `# ignore` line and it doesn't appear supported at all 😢 :https://github.com/bazelbuild/buildtools/issues/890. So sorry about this, I feel like this should be something we could configure.

**Comment by rgkimball** [comment]
Had a feeling you might say that, fixed in the last couple commits per your comments. Should be good to go now!

**Review by betterengineering — APPROVED** [review]

**Pull request merged**